scipy/spatial/_qhull.pyx

```diff
@@ -1435,9 +1435,9 @@ cdef int _find_simplex_directed(DelaunayInfo_t *d, double *c,
                 # The target point is in the direction of neighbor `k`!
                 m = d.neighbors[(ndim+1)*isimplex + k]
                 if m == -1:
-                    # The point is outside the triangulation: bail out
-                    start[0] = isimplex
-                    return -1
+                    # gh-21279
+                    isimplex = _find_simplex_bruteforce(d, c, x, eps, eps_broad)
+                    break
```

Comment:
For reference, if we only keep the `break` and remove the brute force line above, the new test I added does still pass, but `test_ndgriddata.py::TestGriddata::test_fill_value` fails.